components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -333,10 +333,7 @@ void CosmeticFiltersJSHandler::ApplyRules(bool de_amp_enabled) {
         blink::BackForwardCacheAware::kAllow);
   }
 
-  if (!render_frame_->IsMainFrame())
-    return;
-
-  // Working on css rules, we do that on a main frame only
```

Comment:
@SergeyZhukovsky I'm not sure how this happened, but cosmetic filters should be applied in all frames. uBO uses the same approach.

I also added a new test for that behavior.

Reply:
the old extension script injected it only to the main frame iirc.

Reply:
review of old conversations indicates that we took that approach conservatively and would wait to find places where it's required before changing it.

> @pes10k: I dont think it mattes much either way, i bet its way out in the tails to want to hide stuff in embedded frames (instead of just blocking / hiding the frame itself)
> so, just to keep it one issue at a time,  why dont we get andorid to make desktop here, and then we can revisit whether what desktop does is correct later?

https://bravesoftware.slack.com/archives/G019M1QR945/p1600962122004000

I'd say the linked issue + uBO's behavior is reason enough to change it now.

Reply:
I support this fwiw, i think this is a great catch @antonok-edm